theano/tensor/signal/downsample.py

```diff
@@ -834,7 +968,7 @@ def __init__(self, ds, ignore_border, st=None, padding=(0,0), mode='max'):
 
     def make_node(self, x, maxout, gz):
         # make_node should only be called by the grad function of
-        # DownsampleFactorMaxGrad, so these asserts should not fail.
+        # MaxPoolGrad, so these asserts should not fail.
         assert isinstance(x, Variable) and x.ndim == 4
         assert isinstance(maxout, Variable) and maxout.ndim == 4
         assert isinstance(gz, Variable) and gz.ndim == 4
```

Comment:
Can you move the assert after the call to as_tensor_variable?, Then you can only assert ndim, and remove the isinstance.